packages/router/src/lib/operations/auction.ts

```diff
@@ -185,8 +173,75 @@ export const newAuction = async (
     });
   }
 
+  let sendingChainIdx: number = -1;
+  let receivingChainIdx: number = -1;
+  let swapPoolIndex: number = -1;
+  const allowedSwap = config.swapPools.find((pool, index) => {
+    const existSwap =
+      pool.assets.find((a) => getAddress(a.assetId) === getAddress(sendingAssetId) && a.chainId === sendingChainId) &&
+      pool.assets.find((a) => getAddress(a.assetId) === getAddress(receivingAssetId) && a.chainId === receivingChainId);
+
+    if (existSwap) {
+      sendingChainIdx = pool.assets.findIndex(
+        (a) => getAddress(a.assetId) === getAddress(sendingAssetId) && a.chainId === sendingChainId,
+      );
+
+      receivingChainIdx = pool.assets.findIndex(
+        (a) => getAddress(a.assetId) === getAddress(receivingAssetId) && a.chainId === receivingChainId,
+      );
+      swapPoolIndex = index;
+    }
+
+    return existSwap;
+  });
+  if (!allowedSwap) {
+    throw new SwapInvalid(sendingChainId, sendingAssetId, receivingChainId, receivingAssetId, {
+      methodContext,
+      requestContext,
+    });
+  }
+
+  const swapPool = config.swapPools[swapPoolIndex];
+  const routerBalances = await Promise.all(
+    swapPool.assets.map(async (asset) => {
+      const assetLiquidity = await contractReader.getAssetBalance(asset.assetId, asset.chainId);
+      let assetDecimals = chainData.get(asset.chainId.toString())?.assetId[asset.assetId]?.decimals;
+      if (!assetDecimals) {
+        assetDecimals = await txService.getDecimalsForAsset(asset.chainId, asset.assetId);
+      }
+      // convert asset liquidity into 18 decimal value.
+      const res = assetLiquidity.mul(BigNumber.from(10).pow(18 - assetDecimals));
+      return res;
+    }),
+  );
+
+  logger.info("Got balances of router", requestContext, methodContext, {
+    routerBalances: routerBalances.map((balance) => balance.toString()),
+    sendingChainIdx,
+    receivingChainIdx,
+  });
+
+  const [senderLiquidity, receiverLiquidity] = await Promise.all([
+    contractReader.getAssetBalance(sendingAssetId, sendingChainId),
+    contractReader.getAssetBalance(receivingAssetId, receivingChainId),
+  ]);
```

Comment:
these will have different decimals than the `routerBalances` values

Reply:
yeah. no problem. routerBalances are used to calculate swapRate using stableMath so its decimal is 18. but receiverLiquidity is used to check if receiver liquidity is enough to process request or not so its decimal should be equal to receiving asset decimals, any mistakes here?

Reply:
Ahh my mistake -- I thought you were comparing it to `routerBalances` for some reason. Must've misread.

Still, I think we should keep all conversions to and from 18 decimals within the `getReceiverAmount` function to avoid these types of confusions. I was totally wrong about the error here, but since it only has to be 18 in all cases within that function, we should encapsulate all decimal-based manipulation to that function